src/test/java/io/cdap/cdap/events/process/pubsub/PubSubEventWriterTest.java

```diff
@@ -0,0 +1,188 @@
+
+        eventWriter.initialize(mockContext);
+
+        String stringEvent = gson.toJson(mockedEvent);
+        ByteString data = ByteString.copyFromUtf8(stringEvent);
+        PubsubMessage pubsubMessage = PubsubMessage.newBuilder()
+                .setData(data)
+                .build();
+
+
+        eventWriter.publishEvent(mockedEvent);
+        Mockito.verify(mockedPublisher).publish(pubsubMessage);
+    }
+
+    @Test
+    public void testPublishEventKo() {
+        Event mockedEvent = new Event() {
+            @Override
+            public EventType getType() {
+                return EventType.PROGRAM_STATUS;
+            }
+
+            @Override
+            public long getPublishTime() {
+                return 1;
+            }
+
+            @Override
+            public String getVersion() {
+                return "1.0.0";
+            }
+
+            @Override
+            public String getInstanceName() {
+                return "test-instance";
+            }
+
+            @Override
+            public Object getEventDetails() {
+                return "this-is-a-test";
+            }
+        };
+
+        PubSubEventWriter original = new PubSubEventWriter();
+        PubSubEventWriter eventWriter = Mockito.spy(original);
+        Publisher mockedPublisher = Mockito.mock(Publisher.class);
+        when(mockedPublisher.publish(any()))
+                .thenReturn(ApiFutures.immediateFailedFuture(new Exception("Error publishing")));
+        Mockito.doReturn(mockedPublisher).when(eventWriter).getPublisher();
+        EventWriterContext mockContext = () -> mockedProperties;
+
+        eventWriter.initialize(mockContext);
+
+        String stringEvent = gson.toJson(mockedEvent);
+        ByteString data = ByteString.copyFromUtf8(stringEvent);
+        PubsubMessage pubsubMessage = PubsubMessage.newBuilder()
+                .setData(data)
+                .build();
+
+
+        eventWriter.publishEvent(mockedEvent);
+        Mockito.verify(mockedPublisher).publish(pubsubMessage);
+    }
+
+    @Test
+    public void testClose() {
+        PubSubEventWriter original = new PubSubEventWriter();
+        PubSubEventWriter eventWriter = Mockito.spy(original);
+        Publisher mockedPublisher = Mockito.mock(Publisher.class);
+
+        Mockito.doReturn(mockedPublisher).when(eventWriter).getPublisher();
+        Mockito.doNothing().when(mockedPublisher).shutdown();
+        try {
+            Mockito.doReturn(true).when(mockedPublisher).awaitTermination(1, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        EventWriterContext mockContext = () -> mockedProperties;
+
+        eventWriter.initialize(mockContext);
+
+        eventWriter.close();
+        try {
+            Mockito.verify(mockedPublisher).awaitTermination(1, TimeUnit.MINUTES);
+        } catch (Exception e) {
+            assertNull("No exception: ", e);
```

Comment:
Is this explicit null exception check required? The test will fail if there is an exception.